util/Logging.py

```diff
@@ -21,54 +31,60 @@ def limit(_limit_level: int) -> callable:
-                c = conn.cursor()
-                limit_level = c.execute(
-                    "SELECT limitLevel from user where uid = ?",
-                    (author_id,),
-                ).fetchone()
-
-                if limit_level is None:  # User DOES NOT exist in database, add them.
-                    c.execute(
-                        "INSERT INTO user VALUES (?, ?, ?, ?, ?, ?)",
-                        (author_id, 0, False, None, 0, None),
-                    )  # See ERD.mdj
-                    limit_level = 0
-                    conn.commit()
-                else:
-                    limit_level = limit_level[0] or 0
-
-                if limit_level >= _limit_level:
-                    embed = EmbedBuilder(
-                        title="You cannot use this command!",
-                        description="You are limited from using this command! Please contact a moderator if you believe this is a mistake.",
-                        color=0xFF0000,
-                    ).build()
-                    await ctx.respond(embed=embed, ephemeral=True)
-                    Log(
-                        f"$ tried to use {ctx.command.name}. But is limited from using it.",
-                        ctx.author,
-                    )
-                    return None
+                    break
+
+            if not ctx or not ctx.author or not ctx.author.id:
+                # in order to actually do any limiting, we need to know who used the command
+                Log(
+                    f"@limit({limit_level_requirement}) is BROKEN for {func!r}. Failed to find an author!",
+                )
+                return await func(*args, **kwargs)
+
+            conn = sqlite3.connect("util/database.sqlite")
+            c = conn.cursor()
+            limit_level = c.execute(
+                "SELECT limitLevel from user where uid = ?",
+                (ctx.author.id,),
+            ).fetchone()
+
+            if limit_level is None:  # User DOES NOT exist in database, add them.
+                c.execute(
+                    "INSERT INTO user VALUES (?, ?, ?, ?, ?, ?)",
+                    (ctx.author.id, 0, False, None, 0, None),
+                )  # See ERD.mdj
+                limit_level = 0
+                conn.commit()
+            else:
+                limit_level = limit_level[0] or 0
+
+            if limit_level >= limit_level_requirement:
+                embed = EmbedBuilder(
+                    title="You cannot use this command!",
+                    description="You are limited from using this command! Please contact a moderator if you believe this is a mistake.",
+                    color=0xFF0000,
+                ).build()
+                await ctx.respond(embed=embed, ephemeral=True)
+                Log(
+                    f"$ tried to use {ctx.command.name}. But is limited from using it.",
+                    ctx.author,
+                )
+                return None
```

Comment:
This was fine, no need to change this

Reply:
I have removed all changes to util/Logging.py